tests/integration/api/Controller/ShowDiscussionControllerTest.php

```diff
@@ -78,6 +83,51 @@ public function guest_cannot_see_empty_discussion()
         $this->assertEquals(200, $response->getStatusCode());
     }
 
+    /**
+     * @test
+     */
+    public function guest_cannot_see_hidden_posts()
+    {
+        $response = $this->callWith([], ['id' => 4]);
+
+        $json = json_decode($response->getBody()->getContents(), true);
+
+        $this->assertNull(Arr::get($json, 'data.relationships.posts'));
+    }
+
+    /**
+     * @test
+     */
+    public function author_can_see_hidden_posts()
+    {
+        $this->actor = User::find(2);
+
+        $response = $this->callWith([], ['id' => 4]);
+
+        $json = json_decode($response->getBody()->getContents(), true);
+
+        $this->assertEquals(2, Arr::get($json, 'data.relationships.posts.data.0.id'));
+    }
+
+    /**
+     * @test
+     */
+    public function when_allowed_guests_can_see_hidden_posts()
+    {
+        /** @var Dispatcher $events */
+        $events = app(Dispatcher::class);
+
+        $events->listen(ScopeModelVisibility::class, function (ScopeModelVisibility $event) {
+            $event->query->whereRaw('1=1');
+        });
```

Comment:
Shouldn't this listener check for the event's properties so that it only affects the query we want to affect?